### DISCUSSION TITLE
Quay.io backend S3 storage

### DISCUSSION THREAD

**Discussion by BostjanBozic** [quay/quay#1612] [Q&A] [answered]
Hello,

I wonder if someone can point me to documentation (or provide me with a list) regarding Quay's backing S3 buckets?

Reason why I would need this information is that we are using AWS VPC S3 gateway endpoint with Endpoint policy and I need to whitelist quay.io there (traffic goes through VPC S3 endpoint, since CloudFront is pointing toward S3 buckets for image layers.

Thank you and best regards,
Bostjan

**Comment by BillDett** [comment, marked as answer]
Bostjan- we actually document this thru OpenShift: https://docs.openshift.com/container-platform/4.11/installing/install_config/configuring-firewall.html <https://docs.openshift.com/container-platform/4.11/installing/install_config/configuring-firewall.html>

Hope that helps,

Bill Dettelback
He/Him
Sr. Engineering Manager, Quay / OpenShift Registry
Red Hat  <https://www.redhat.com/>
 <https://www.redhat.com/>

**Comment by BostjanBozic** [comment]
Hello Bill,

thank you for fast reply on this. Perfect, for some reason I was going through Quay code and docs, never considered checking OpenShift docs :) Adding `quayio-production-s3` to my VPC S3 endpoint policy resolved the issue.

Thank you again!

**Comment by prashant0085** [comment]
@BostjanBozic how did you added the bucket in your policy? by following name `quayio-production-s3.s3.amazonaws.com` or s3 arn like this `arn:aws:s3:::quayio-production-s3/*`.

Please let me know.

Thanks

**Comment by BostjanBozic** [comment]
@prashant0085 I added is as following (added some other ones there as well to cover DockerHub, gcr.io and some others (GetObject permission is added there):

```
"arn:aws:s3:::docker-images-prod/*",
"arn:aws:s3:::jfrog-prod-use1-shared-virginia-main/*",
"arn:aws:s3:::prod-registry-k8s-io-*",
"arn:aws:s3:::prod-${var.aws_region}-starport-layer-bucket/*",
"arn:aws:s3:::quayio-production-s3/*",
"arn:aws:s3:::repo.${var.aws_region}.amazonaws.com/*"
```

**Reply by prashant-tala** [reply]
@BostjanBozic Thanks,

I had added starport and k8s registry but quay and docker s3 bucket were missing, Adding them fixed the issue.
